src/hdmf_zarr/backend.py

```diff
@@ -161,7 +162,7 @@ def open(self):
             # As a result, when in other modes, the file will not use consolidated metadata.
             if self.__mode not in ['r', 'r+']:
                 self.__file = zarr.open(store=self.path,
-                                        mode=self.__mode,
+                                        mode=self.__mode if self.__mode != 'r-' else 'r',
```

Comment:
@oruebel I think it is cleaner to not have this conditional statement in a function call. Could you move it out?

Reply:
Fixed